cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -39,31 +40,45 @@ namespace cuvs::neighbors::cagra {
  */
 
 /**
- * @brief ANN algorithm used by CAGRA to build knn graph
+ * @brief ANN parameters used by CAGRA to build knn graph
  *
  */
-enum class graph_build_algo {
-  /* Use IVF-PQ to build all-neighbors knn graph */
-  IVF_PQ,
-  /* Experimental, use NN-Descent to build all-neighbors knn graph */
-  NN_DESCENT
+struct graph_build_params {
+  bool construct_index_with_dataset = true;
+  virtual ~graph_build_params() {}
 };
 
+namespace ivf_pq {
+/** Specialized parameters utilizing IVF-PQ to build knn graph */
+struct graph_build_params : cuvs::neighbors::cagra::graph_build_params {
+  std::optional<cuvs::neighbors::ivf_pq::index_params> build_params   = std::nullopt;
```

Comment:
This is a bit awkward to have optional here. A struct that could be potentially empty? I think we should remove the std::optional and make these reasonable defaults.